BDArmory/UI/BDAEditorArmorWindow.cs

```diff
@@ -1123,6 +1175,290 @@ void Visualize()
+                                        {
+                                            nonRootCockpit = true;
+                                        }
+                                        */
+                                        break;
+                                    }
+                            }
+                        }
+                    }
+                    if (BDArmorySettings.MAX_PWING_LIFT > 0)
+                    {
+                        foreach (var part in EditorLogic.fetch.ship.Parts) //not ideal, but this needs to fire onVesselModified, not onPartPlaced, but linking this to Visualizer's parts eval only updates when that does
+                        {
+                            if (part.name.Contains("B9.Aero.Wing.Procedural.Type"))
+                            {
+                                ModuleLiftingSurface wing = part.GetComponent<ModuleLiftingSurface>();
+                                if (wing != null && wing.deflectionLiftCoeff > 0f)
+                                {
+                                    if (wing.deflectionLiftCoeff > BDArmorySettings.MAX_PWING_LIFT)
+                                        oversizedPWings++;
+                                }
+                            }
+                        }
+                    }
+                }
+                StringBuilder evaluationstring = new StringBuilder();
+                ScreenMessage vessellegality = new ScreenMessage("", 7.0f, ScreenMessageStyle.LOWER_CENTER);
+                if (CompSettings.CompVesselChecksEnabled)
+                {
+                    if (maxPartCount > 0 && EditorLogic.fetch.ship.Parts.Count > maxPartCount)
+                        evaluationstring.AppendLine($"{StringUtils.Localize("#LOC_BDArmory_ArmorToolPartCount")} ({EditorLogic.fetch.ship.Parts.Count}/{maxPartCount})");
+                    if (engineCount > 0)
+                    {
+                        if (maxEngines >= 0 && engineCount > maxEngines)
+                            evaluationstring.AppendLine($"{StringUtils.Localize("#LOC_BDArmory_ArmorToolEngineCount")} ({engineCount}/{maxEngines}) - {engineparts}");
+                        if (maxEngines < 0 && engineCount < Mathf.Abs(maxEngines))
+                            evaluationstring.AppendLine($"{StringUtils.Localize("#LOC_BDArmory_ArmorToolEngineCountFloor")} ({engineCount}/{Mathf.Abs(maxEngines)})");
+                    }
+                    if (maxTWR > 0 && ((maxThrust / (PhysicsGlobals.GravitationalAcceleration * FlightGlobals.GetHomeBody().GeeASL) * EditorLogic.fetch.ship.GetTotalMass())) > maxLtW)
+                        evaluationstring.AppendLine($"{StringUtils.Localize("#LOC_BDArmory_ArmorToolTWR")} {Math.Round(maxThrust / (EditorLogic.fetch.ship.GetTotalMass() * (PhysicsGlobals.GravitationalAcceleration * FlightGlobals.GetHomeBody().GeeASL)), 2)}/{maxTWR}");
+                    if (maxLtW > 0 && wingLoadingWet > maxLtW)
```

Comment:
Several of these (like `wingLoadingWet`) aren't updated unless the armour GUI is open.
The functions that update these will likely need to be called at the start of the `if (CompSettings.CompVesselChecksEnabled)` block.